### PR TITLE
New extrapolation process to extrapolate in betweeen known values as well

### PR DESCRIPTION
**Pull request by garypolzin**
# Description
The outcome of the extrapolation process matches the current process, however the changes made mean that we have forward extrapolated values in between data points which we need for the interpolation process.

Currently this code isn't being used by the pipeline.

# How to test
Test data passing and behaving as expected

# Developer checklist
- [X] Unit test
- [X] Linked to Trello ticket
- [X] Documentation up to date

### CODE REVIEW

File: tests/test_file_data.py [hunk 1]
```diff
@@ -3862,22 +3862,176 @@ class ModelExtrapolationAndInterpolation:
...
+        ("1-005", date(2023, 9, 1), 1693526400, None, PrimaryServiceType.non_residential, 1300.0),
+        ("1-005", date(2023, 10, 1), 1696118400, None, PrimaryServiceType.non_residential, 1350.0),
+        ("1-005", date(2023, 11, 1), 1698796800, None, PrimaryServiceType.non_residential, 1400.0),
+        ("1-005", date(2023, 12, 1), 1701388800, None, PrimaryServiceType.non_residential, 1450.0),
+        ("1-005", date(2024, 1, 1), 1704067200, None, PrimaryServiceType.non_residential, 1400.0),
+        ("1-005", date(2024, 2, 1), 1706745600, None, PrimaryServiceType.non_residential, 1350.0),
+        ("1-005", date(2024, 3, 1), 1709251200, 5.0, PrimaryServiceType.non_residential, 1300.0),
+        ("1-005", date(2024, 4, 1), 1711929600, None, PrimaryServiceType.non_residential, 1380.0),
+        ("1-005", date(2024, 5, 1), 1714521600, 15.0, PrimaryServiceType.non_residential, 1400.0),
+        ("1-005", date(2024, 6, 1), 1717200000, None, PrimaryServiceType.non_residential, 1300.0),
```
Comment (garypolzin) [inline tests/test_file_data.py:3890]: this additional data is converted from Will's excel spreadsheet - I'll probably delete it at the end of the process as it's not adding anything to the tests other than checking outputs match excel

Comment (MarionJHolloway) [reply tests/test_file_data.py:3890]: Maybe add a comment to the code to this effect?

File: utils/estimate_filled_posts/models/extrapolation_new.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+    return df
+
+
+def define_window_specs() -> Tuple[Window, Window]:
+    """
+    Defines a window specification which is partitioned by 'location_id' and ordered by 'unix_time'.
+
+    Returns:
+        Window: The window specification partitioned by 'location_id' and ordered by 'unix_time'.
```
Comment (MarionJHolloway) [inline utils/estimate_filled_posts/models/extrapolation_new.py:52]: This returns a tuple of windows, not a window

**Review by MarionJHolloway — APPROVED** [review]
Just a couple of small notes

File: utils/estimate_filled_posts/models/extrapolation_new.py (hunk 2 above)
Comment (garypolzin) [reply utils/estimate_filled_posts/models/extrapolation_new.py]: good spot, forgot to update the docstring from the earlier version

**Pull request merged**